authors/apps/articles/models.py

```diff
@@ -0,0 +1,69 @@
+import random
+import string
+
+from django.db import models
+
+# Create your models here.
+from django.template.defaultfilters import slugify
+
+from authors.apps.core.models import BaseModel
+
+
+class Article(BaseModel):
+    """
+    Model for an article, extends a basemodel since the created and updated times are required
+    """
+    slug = models.SlugField(max_length=255, unique=True, db_index=True)
+    title = models.CharField(max_length=255)
+    description = models.TextField()
+    body = models.TextField()
+    image = models.URLField(default='')
+
+    # The author Foreign Key should be bound to a user profile
+    author = models.ForeignKey(
+        'authentication.User',
+        related_name='articles',
+        on_delete=models.CASCADE
+    )
+    # a article contains many tags
+    tags = models.ManyToManyField(
+        'articles.Tag',
+        related_name='articles',
+    )
+    published = models.BooleanField(default=False)
+
+    def save(self, *args, **kwargs):
```

Comment:
This works but I would encourage the use of `pre_save` signal instead.

Reply:
Cool. Let me do some research on how to use the `pre_save` so that I can change it.

Reply:
From the DRF documentation, I hear the `pre_save`, `post_save`, `pre_delete`, `post_delete` methods are deprecated. 
https://www.django-rest-framework.org/api-guide/generic-views/

Reply:
`DRF pre_save, post_save, pre_delete, post_delete` methods are different from  Django methods. `DRF pre_save, post_save, pre_delete, post_delete` act on `View` layer while the ones for Django act on the model layer which is what we want.

Reply:
Oh. Cool. Thanks for the clarification.
Let me work on it.